.github/workflows/main.yaml

```diff
@@ -0,0 +1,42 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - 'master'
+  pull_request:
+    branches:
+      - '*'
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  # Go needs absolute directories, using the $HOME variable doesn't work here.
+  GOPATH: /home/runner/work/go
+
+  GO_VERSION: '1.22.11'
+
+jobs:
+  rust-compilation-check:
+    name: Rust compilation check check
+    runs-on: ubuntu-latest
+    steps:
+      - name: build wrapper
+        run: make build-wrapper
```

Comment:
Can we add a check here that wrapper.wasm was not changed by this, to ensure the comitted wrapper.wasm file is always up to date?